compiler/passes/ResolveScope.cpp

```diff
@@ -1140,10 +1142,46 @@ Symbol* ResolveScope::getFieldLocally(const char* fieldName) const {
 ************************************** | *************************************/
 
 bool ResolveScope::matchesTypeWithMethods(const char* name) const {
-  std::set<const char*>::const_iterator it = mMethodsOnTypeName.find(astr(name));
   // Returns true if we found a method defined on this name in scope, false
   // if no such method was found
-  return it != mMethodsOnTypeName.end();
+
+  auto it = mMethodsOnTypeName.find(astr(name));
+  if (it != mMethodsOnTypeName.end()) {
+    return true;
+  }
+
+  // If it doesn't re-export, no point checking public uses for matching methods.
+  if (!this->canReexport) return false;
+
+  for_vector_allowing_0s(VisibilityStmt, visStmt, mUseImportList) {
+    // Note: assumes that UseStmt and ImportStmt are the only subclasses of
+    // VisibilityStmt
+    if (visStmt != NULL) {
+      if (!visStmt->isPrivate) {
+        if (!visStmt->skipSymbolSearch(name)) {
+          const char *nameToUse = name;
+          const bool isSymRenamed = visStmt->isARenamedSym(name);
+          if (isSymRenamed) {
+            nameToUse = visStmt->getRenamedSym(name);
+          }
+          if (SymExpr *se = toSymExpr(visStmt->src)) {
+            if (ModuleSymbol *ms = toModuleSymbol(se->symbol())) {
+              ResolveScope *scope = ResolveScope::getScopeFor(ms->block);
+
+              // Mimick the behavior of lookupPublicUnqualAccessSyms, and don't
+              // recurse. Maybe not sufficient?
+              auto it = scope->mMethodsOnTypeName.find(astr(nameToUse));
```

Comment:
Why not recurse? Recursing seems appropriate here but you'd need to pass around a set of scopes already visited & not re-visit.

Reply:
Well, I think the behavior of this and `lookupPublicUnqualAccessSyms` should match (otherwise, tertiary method imports work differently from regular symbol imports?). Of course, I could refactor `lookupPublicUnqualAccessSyms` to also recurse, but I think that it's best to leave that for another PR (which would fix both occurrences of non-recursing, and address #22749)